src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.cs

```diff
@@ -0,0 +1,96 @@
+
+        public override ImmutableArray<string> FixableDiagnosticIds
+        {
+            get
+            {
+                return ImmutableArray.Create(CS0108);
+            }
+        }
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            var diagnostic = context.Diagnostics.First();
+            var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+            var token = root.FindToken(diagnosticSpan.Start);
+            SyntaxNode originalNode = token.GetAncestor<PropertyDeclarationSyntax>();
+            
+            if (originalNode == null)
+            {
+                originalNode = token.GetAncestor<MethodDeclarationSyntax>();
+            }
+
+            if(originalNode == null)
+            {
+                originalNode = token.GetAncestor<FieldDeclarationSyntax>();
+            }
+
+            if(originalNode == null)
+            {
+                return;
+            }
+
+            var newNode = GetNewNode(context.Document, originalNode, context.CancellationToken);
+
+            if (newNode == null)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(new AddNewKeywordAction(context.Document, originalNode, newNode), context.Diagnostics);
+        }
+
+        private SyntaxNode GetNewNode(Document document, SyntaxNode node, CancellationToken cancellationToken)
+        {
+            SyntaxNode newNode = null;
+
+            var propertyStatement = node as PropertyDeclarationSyntax;
+            if (propertyStatement != null)
+            {
+                newNode = propertyStatement.AddModifiers(SyntaxFactory.Token(SyntaxTriviaList.Empty, SyntaxKind.NewKeyword, SyntaxTriviaList.Create(SyntaxFactory.Whitespace(" ")))) as SyntaxNode;
+            }
+
+            var methodStatement = node as MethodDeclarationSyntax;
+            if (methodStatement != null)
+            {
+                newNode = methodStatement.AddModifiers(SyntaxFactory.Token(SyntaxTriviaList.Empty, SyntaxKind.NewKeyword, SyntaxTriviaList.Create(SyntaxFactory.Whitespace(" "))));
+            }
+
+            var fieldDeclaration = node as FieldDeclarationSyntax;
+            if (fieldDeclaration != null)
+            {
+                newNode = fieldDeclaration.AddModifiers(SyntaxFactory.Token(SyntaxTriviaList.Empty, SyntaxKind.NewKeyword, SyntaxTriviaList.Create(SyntaxFactory.Whitespace(" "))));
+            }
+
+            var cleanupService = document.GetLanguageService<ICodeCleanerService>();
+
+            if (cleanupService != null && newNode != null)
+            {
+                newNode = cleanupService.Cleanup(newNode, new[] { newNode.Span }, document.Project.Solution.Workspace, cleanupService.GetDefaultProviders(), cancellationToken);
+            }
```

Comment:
It is not necessary to run the ICodeCleanerService.  You can just add the [Formatter.Annotation](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Formatting/Formatter.cs,24) to the node you return, and the node will be formatted with the users formatting options automatically when the formatter runs.
example:
`return newNode.WithAdditionalAnnotations(Formatter.Annotation);`